trace/opencensus/trace/samplers/fixed_rate.py

```diff
@@ -12,30 +12,46 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-import random
-
 from opencensus.trace.samplers.base import Sampler
 
 DEFAULT_SAMPLING_RATE = 0.5
 
+MAX_VALUE = 0xffffffffffffffff
+
 
 class FixedRateSampler(Sampler):
     """Sample a request at a fixed rate.
 
     :type rate: float
     :param rate: The rate of sampling.
     """
-    def __init__(self, rate=DEFAULT_SAMPLING_RATE):
+    def __init__(self, rate=None):
+        if rate is None:
+            rate = DEFAULT_SAMPLING_RATE
+
         if rate > 1 or rate < 0:
             raise ValueError('Rate must between 0 and 1.')
 
         self.rate = rate
 
-    @property
-    def should_sample(self):
-        random_number = random.uniform(0, 1)
+    def should_sample(self, trace_id):
+        lower_long = get_lower_long_from_trace_id(trace_id)
+        bound = self.rate * MAX_VALUE
 
-        if random_number <= self.rate:
+        if lower_long <= bound:
```

Comment:
Do we know that this the same logic that Java et al uses?  Has this been documented in the specs repo yet?

Reply:
Yes, but as there is no max value for long type in Python, I use the max value for a 8 bytes hex as the max value.